vcr/holder/oidc4vci_wallet.go

```diff
@@ -113,53 +113,60 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
 	}
 	issuerClient, err := h.issuerClientCreator(ctx, httpClient, offer.CredentialIssuer)
 	if err != nil {
-		return fmt.Errorf("unable to create issuer client: %w", err)
+		return oidc4vci.Error{
+			Err:        fmt.Errorf("unable to create issuer client: %w", err),
+			Code:       oidc4vci.ServerError,
+			StatusCode: http.StatusInternalServerError,
+		}
 	}
 
-	// TODO: store offer and perform these requests async
-	//       See https://github.com/nuts-foundation/nuts-node/issues/2040
 	accessTokenResponse, err := issuerClient.RequestAccessToken(oidc4vci.PreAuthorizedCodeGrant, map[string]string{
 		"pre-authorized_code": preAuthorizedCode,
 	})
 	if err != nil {
-		return fmt.Errorf("unable to request access token: %w", err)
+		return oidc4vci.Error{
+			Err:        fmt.Errorf("unable to request access token: %w", err),
+			Code:       oidc4vci.InvalidToken,
```

Comment:
it is weird to receive a InvalidToken error on credential offer, but probably fine in a temporary situation?